src/openfl/display/DisplayObject.hx

```diff
@@ -1325,26 +1320,92 @@ class DisplayObject extends EventDispatcher implements IBitmapDrawable #if openf
 	}
 	
 	
-	public function __updateChildren (transformOnly:Bool):Void {
+	private function __updateRenderData ():Void {
 		
 		var renderParent = __renderParent != null ? __renderParent : parent;
+		if (__isMask && renderParent == null) renderParent = __maskTarget;
 		__renderable = (visible && __scaleX != 0 && __scaleY != 0 && !__isMask && (renderParent == null || !renderParent.__isMask));
-		__worldAlpha = alpha;
-		__worldBlendMode = blendMode;
 		
-		if (__transformDirty) {
+		__updateTransforms ();
+		
+		if (__supportDOM) {
+				
+				__renderTransformChanged = !__renderTransform.equals (__renderTransformCache);
+				
+				if (__renderTransformCache == null) {
+					
+					__renderTransformCache = __renderTransform.clone ();
+					
+				} else {
+					
+					__renderTransformCache.copyFrom (__renderTransform);
+					
+				}
+				
+			}
 			
-			__transformDirty = false;
+			if (!__worldColorTransform.__equals (transform.colorTransform)) {
+				
+				__worldColorTransform = transform.colorTransform.__clone ();
+				
+			}
 			
-		}
+			if (renderParent != null) {
+				
+				if (__supportDOM) {
+					
+					var worldVisible = (renderParent.__worldVisible && visible);
+					__worldVisibleChanged = (__worldVisible != worldVisible);
+					__worldVisible = worldVisible;
+					
+					var worldAlpha = alpha * renderParent.__worldAlpha;
+					__worldAlphaChanged = (__worldAlpha != worldAlpha);
+					__worldAlpha = worldAlpha;
+					
+				} else {
+					
+					__worldAlpha = alpha * renderParent.__worldAlpha;
+					
+				}
+				
+				__worldColorTransform.__combine (renderParent.__worldColorTransform);
+				
+				if (__blendMode == null || __blendMode == NORMAL) {
+					
+					// TODO: Handle multiple blend modes better
+					__worldBlendMode = renderParent.__blendMode;
+					
+				} else {
+					
+					__worldBlendMode = __blendMode;
+					
+				}
+				
+			} else {
+				
+				__worldAlpha = alpha;
+				
+				if (__supportDOM) {
+					
+					__worldVisibleChanged = (__worldVisible != visible);
+					__worldVisible = visible;
+					
+					__worldAlphaChanged = (__worldAlpha != alpha);
+					
+				}
```

Comment:
I still think, when we already change all this now, we should change it for DOM as well. The problem is that `__worldAlphaChanged` is always true here. Maybe this is intended, but by looking at the code with the `renderParent` above I'm not so sure about that. I would do the following:

```
if (__supportDOM) {

	__worldVisibleChanged = (__worldVisible != visible);
	__worldVisible = visible;

	__worldAlphaChanged = (__worldAlpha != alpha);
	__worldAlpha = alpha;

} else {

	__worldAlpha = alpha;

}
```